src/libgriddyn/griddyn/bits/griddyn_types.h

```diff
@@ -0,0 +1,26 @@
+#ifndef LIBGRIDDYN_BITS_GRIDDYN_TYPES_H
+#define LIBGRIDDYN_BITS_GRIDDYN_TYPES_H
+
+#include <griddyn/bits/griddyn_structs.h>
+
+#include <stddef.h> // size_t
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef size_t    griddyn_idx_t;
+typedef double    griddyn_value_t;
+typedef int       griddyn_bool_t;
+
+typedef struct griddyn_sim_   griddyn_sim;
+typedef struct griddyn_bus_   griddyn_bus;
+typedef struct griddyn_link_  griddyn_link;
+typedef struct griddyn_load_  griddyn_load;
+typedef struct griddyn_generator_ griddyn_generator;
+
```

Comment:
area and relay would also seem to relevent at this level?

Reply:
Relay, sure, but none of the models I've seen actually have relays in them. Area's functionality doesn't seem very useful, considering how much of a headache another layer of mapping (like generators/loads, except backwards I think?) it would be.